### PR TITLE
Metric formatting

### PR DESCRIPTION
**Pull request by cikzh**
Adds Prometheus formatting for most dataset members. Builds on #390

### CODE REVIEW

**Comment by cikzh** [comment]
@davidv1992 I've pulled the formatting code into it's own file, since it was getting quite long. While reviewing, please pay attention to the correctness of the descriptions given in the `format_metric` calls.

File: statime-linux/src/metrics/exporter.rs [hunk 1]
```diff
@@ -1,20 +1,16 @@
-use std::{
-    fmt::Write,
-    path::{Path, PathBuf},
-};
+use std::path::{Path, PathBuf};
 
 use clap::Parser;
 use serde::{Deserialize, Serialize};
+use statime::observability::ObservableInstanceState;
```
Comment (davidv1992) [inline statime-linux/src/metrics/exporter.rs:5]: As discussed in person, can you remove this struct and associated functions from statime.

File: statime/src/datastructures/datasets/parent.rs [hunk 2]
```diff
@@ -5,9 +5,9 @@ use crate::datastructures::common::{ClockIdentity, ClockQuality, PortIdentity};
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub(crate) struct InternalParentDS {
     pub(crate) parent_port_identity: PortIdentity,
-    pub(crate) parent_stats: bool,
-    pub(crate) observed_parent_offset_scaled_log_variance: u16,
-    pub(crate) observed_parent_clock_phase_change_rate: u32,
+    //pub(crate) parent_stats: bool,
+    //pub(crate) observed_parent_offset_scaled_log_variance: u16,
+    //pub(crate) observed_parent_clock_phase_change_rate: u32,
```
Comment (davidv1992) [inline statime/src/datastructures/datasets/parent.rs:10]: Why not just remove this

File: statime/src/datastructures/datasets/parent.rs [hunk 3]
```diff
@@ -21,9 +21,9 @@ impl InternalParentDS {
                 clock_identity: default_ds.clock_identity,
                 port_number: 0,
             },
-            parent_stats: false,
-            observed_parent_offset_scaled_log_variance: 0xffff,
-            observed_parent_clock_phase_change_rate: 0x7fffffff,
+            //parent_stats: false,
+            //observed_parent_offset_scaled_log_variance: 0xffff,
+            //observed_parent_clock_phase_change_rate: 0x7fffffff,
```
Comment (davidv1992) [inline statime/src/datastructures/datasets/parent.rs:26]: Why not just remove this

File: statime/src/observability/current.rs [hunk 4]
```diff
@@ -2,23 +2,26 @@ use crate::datastructures::datasets::InternalCurrentDS;
...
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct CurrentDS {
     /// See *IEEE1588-2019 section 8.2.2.2*.
     pub steps_removed: u16,
     /// See *IEEE1588-2019 section 8.2.2.3*.
     pub offset_from_master: i128,
-    /// See *IEEE1588-2019 section 8.2.2.4*.
-    pub mean_delay: i128,
+    // /// See *IEEE1588-2019 section 8.2.2.4*.
+    // pub mean_delay: i128,
```
Comment (davidv1992) [inline statime/src/observability/current.rs:16]: Why not just remove this.

File: statime/src/observability/current.rs [hunk 5]
```diff
@@ -2,23 +2,26 @@ use crate::datastructures::datasets::InternalCurrentDS;
...
 }
 
 impl From<&InternalCurrentDS> for CurrentDS {
     fn from(v: &InternalCurrentDS) -> Self {
         Self {
             steps_removed: v.steps_removed,
             offset_from_master: v.offset_from_master.nanos_rounded(),
-            mean_delay: v.mean_delay.nanos_rounded(),
+            // mean_delay: v.mean_delay.nanos_rounded(),
```
Comment (davidv1992) [inline statime/src/observability/current.rs:24]: Why not just remove this.

File: statime/src/observability/parent.rs [hunk 6]
```diff
@@ -3,19 +3,22 @@ use crate::{
...
     pub parent_port_identity: PortIdentity,
-    /// See *IEEE1588-2019 section 8.2.3.3*.
-    pub parent_stats: bool,
-    /// See *IEEE1588-2019 section 8.2.3.4*.
-    pub observed_parent_offset_scaled_log_variance: u16,
-    /// See *IEEE1588-2019 section 8.2.3.5*.
-    pub observed_parent_clock_phase_change_rate: u32,
+    // pub parent_stats: bool,
+    // pub observed_parent_offset_scaled_log_variance: u16,
+    // pub observed_parent_clock_phase_change_rate: u32,
```
Comment (davidv1992) [inline statime/src/observability/parent.rs:21]: why not just remove this?

File: statime/src/observability/parent.rs [hunk 7]
```diff
@@ -30,10 +33,10 @@ impl From<&InternalParentDS> for ParentDS {
...
         Self {
             parent_port_identity: v.parent_port_identity,
-            parent_stats: v.parent_stats,
-            observed_parent_offset_scaled_log_variance: v
-                .observed_parent_offset_scaled_log_variance,
-            observed_parent_clock_phase_change_rate: v.observed_parent_clock_phase_change_rate,
+            // parent_stats: v.parent_stats,
+            // observed_parent_offset_scaled_log_variance: v
+            //     .observed_parent_offset_scaled_log_variance,
+            // observed_parent_clock_phase_change_rate: v.observed_parent_clock_phase_change_rate,
```
Comment (davidv1992) [inline statime/src/observability/parent.rs:39]: Why not just remove this

**Review by davidv1992 — CHANGES_REQUESTED** [review]
Some minor things around commenting out lines instead of outright removing em.

**Review by davidv1992 — APPROVED** [review]

**Pull request merged**